packages/esm-patient-tests-app/src/edit-test-results/modal/edit-lab-results.modal.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import { Button, ModalBody, ModalFooter, ModalHeader, RadioButton, RadioButtonGroup } from '@carbon/react';
+import { formatDate, launchWorkspace, useSession } from '@openmrs/esm-framework';
+import { type Order } from '@openmrs/esm-patient-common-lib';
+import styles from './edit-lab-results.scss';
+
+type EditLabResultModalProps = {
+  orders: Array<Order>;
+  closeModal: () => void;
+};
+
+const EditLabResultModal: React.FC<EditLabResultModalProps> = ({ orders, closeModal }) => {
+  const { t } = useTranslation();
+  const [selectedOrder, setSelectedOrder] = useState<string>(orders?.[0]?.uuid);
+  const { sessionLocation } = useSession();
+  const location = sessionLocation?.display;
+
+  const handleOrderSelection = (orderId: string) => {
+    setSelectedOrder(orderId);
+  };
+
+  const filteredOrders = useMemo(() => {
+    return orders.filter((order) => order.uuid === selectedOrder);
+  }, [orders, selectedOrder]);
```

Comment:
We don't need to run a useMemo to find the selected order from the orders list every time the user updates the selected order, we can pass in the order object to the `handleOrderSelection` function and have a state to store the selected order object. Or you could change the `selectedOrder` state to store the order object instead of just the order UUID.